File: contents/02/extra-1/solution.cpp
#include<iostream>
#include<string>
#include<stack>

using namespace std;

int main() {
	int N, K;
	string S;
	cin >> N >> K >> S;
	int cnt = 0;
	stack<int>st;
	for (int i = 0; i < N; i++) {
		if (S[i] == '(')st.push(i);
		else st.pop();
		if (st.empty())cnt++;
	}
	int ans = K / cnt;
	cout << ans << endl;
	return 0;
}

File: contents/02/extra-1/solution.py
N, K = map(int, input().split())
S = input()

stack = []
cnt = 0
for i in range(len(S)):
    if S[i]=='(':
        stack.append(S[i])
    else:
        stack.pop()
    if len(stack) == 0:
        cnt += 1

print(K//cnt)